### PR TITLE
Minor code quality improvements

### PR DESCRIPTION
**Pull request by adams85**
### Describe the purpose of your pull request

While familiarizing myself with the SDK, I found some rough edges in the implementation (at least, I think so). Each commit in this PR proposes a separate minor fix/improvement/addition described by the commit message.

A few additional notes on some of the commits:
* https://github.com/configcat/.net-sdk/commit/54c613023188e30841db9d0a5031ee251506021c: I'd even suggest using `TaskCompletionSource<T>` instead of `ManualResetEventSlim` because that's a better fit for the async/await programming model and doesn't use unmanaged resources, however [there's a gotcha around TCSs](https://devblogs.microsoft.com/premier-developer/the-danger-of-taskcompletionsourcet-class/). The fix for this problem (`TaskCreationOptions.RunContinuationsAsynchronously`) is available only since .NET 4.6.1 ([actually, 4.6.2](https://support.microsoft.com/en-us/topic/known-issues-for-the-net-framework-4-6-1-0b42edba-62b9-d479-3da7-1e80af01865a)), however the SDK supports older .NET versions. The problem can be worked around in those old versions as well but that would require some further discussion.
* https://github.com/configcat/.net-sdk/commit/db988389b18081ac59df19346439c2ae10d1de6d: AFAIK, async file I/O was kind of a mess in .NET but [things were improved greatly in .NET 6](https://devblogs.microsoft.com/dotnet/file-io-improvements-in-dotnet-6/). So in the light of that we may even consider an asynchronous read [here](https://github.com/configcat/.net-sdk/blob/db988389b18081ac59df19346439c2ae10d1de6d/src/ConfigCatClient/Override/LocalFileDataSource.cs#L89).
* https://github.com/configcat/.net-sdk/commit/c12e0517804b97a1a03721342f7b5c965b16cb78: This eliminates a bunch of unnecessary memory allocations but newing up a hasher object on every call of `StringExtensions.Hash` still doesn't feel good to me. If you agree, the situation could be easily improved.
* https://github.com/configcat/.net-sdk/commit/17a259bca5ccafc76c3196942aa824bcfa1ab5f1: The sample project is pretty outdated (netcoreapp2.2), it'd be nice to update it to the latest .NET and SDK version.

### Requirement checklist (only if applicable)

- [ ] I have covered the applied changes with automated tests.
- [ ] I have executed the full automated test set against my changes.
- [ ] I have validated my changes against all supported platform versions.
- [ ] I have read and accepted the [contribution agreement](https://github.com/configcat/legal/blob/main/contribution-agreement.md).

### CODE REVIEW

File: src/ConfigCatClient/Configuration/AutoPollConfiguration.cs [hunk 1]
```diff
@@ -38,7 +38,7 @@ internal override void Validate()
         {
             base.Validate();
 
-            if (PollIntervalSeconds == 0)
+            if (PollIntervalSeconds <= 0)
```
Comment (endret) [inline src/ConfigCatClient/Configuration/AutoPollConfiguration.cs:41]: Can you explain why it is necessary to check negative values?

Comment (adams85) [reply src/ConfigCatClient/Configuration/AutoPollConfiguration.cs:41]: `TimeSpan`s are allowed to be negative. For example `TimeSpan.FromSeconds(-2).TotalSeconds` returns -2. I can't imagine that a negative interval makes sense in this case. The exception message says "Value must be **greater than zero**."

BTW, `TotalSeconds` does some unnecessary calculations under the hood, `PollInterval <= TimeSpan.Zero` would be just fine. I'll correct this too.

Comment (endret) [reply src/ConfigCatClient/Configuration/AutoPollConfiguration.cs:41]: TimeSpan could be, but it is the `uint` type. I guess the customer should get a compile-time error.

Comment (adams85) [reply src/ConfigCatClient/Configuration/AutoPollConfiguration.cs:41]: You're right, I missed that `PollIntervalSeconds` was an `uint`. That change is unnecessary indeed. I'll revert it.

On top of that oversight, when I replied to your comment, I confused `PollIntervalSeconds` with `PollInterval.TotalSeconds`. What I wrote applies, of course, to the latter.

**Review by endret — APPROVED** [review]

**Comment by adams85** [comment]
I dropped the problematic commit and rebased it to master, so it's ready for merge.

**Review by z4kn4fein — APPROVED** [review]

**Pull request merged**